src/main/java/org/mythicmc/editstick/util/BlockUtils.java

```diff
@@ -1,29 +1,76 @@
 package org.mythicmc.editstick.util;
 
+import org.bukkit.Axis;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Bisected;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.Directional;
+import org.bukkit.block.data.Orientable;
+import org.bukkit.block.data.type.Slab;
 
 public class BlockUtils {
     public static void rotateBlock(Block block) {
         BlockData blockData = block.getBlockData();
+        //-- Change "Facing"
         if (blockData instanceof Directional) {
             Directional directional = (Directional) blockData;
             BlockFace face = directional.getFacing();
             if (face == BlockFace.NORTH) {
                 directional.setFacing(BlockFace.EAST);
-                block.setBlockData(directional);
             } else if (face == BlockFace.EAST) {
                 directional.setFacing(BlockFace.SOUTH);
-                block.setBlockData(directional);
             } else if (face == BlockFace.SOUTH) {
                 directional.setFacing(BlockFace.WEST);
-                block.setBlockData(directional);
             } else if (face == BlockFace.WEST) {
-                directional.setFacing(BlockFace.NORTH);
-                block.setBlockData(directional);
+                if(directional.getFaces().contains(BlockFace.UP)){
```

Comment:
Slight stylistic change.
```suggestion
                if (directional.getFaces().contains(BlockFace.UP)) {
```